### DISCUSSION TITLE
Some questions about using tobac to track my own case of interest

### DISCUSSION THREAD

**Discussion by ESSTOOL** [tobac-project/tobac#229] [Q&A] [answered]
After successfully trying the case of tracking of deep convection (https://github.com/tobac-project/tobac/blob/hotfix/examples/Example_OLR_Tracking_model/Example_OLR_Tracking_model.ipynb), I have a few questions...

a) I found that the WRF data of the example seems to be specially post-processed (Example_input_OLR_model.nc), because its latitude and longitude variables are "latitude" and "longitude" instead of XLAT and XLONG...When I use my own post-processed WRF output data, running "dxy,dt=tobac.get_spacings(OLR)" will report an error:

ValueError: no information about grid spacing, need either input cube with projection_x_coord and projection_y_coord or keyword argument grid_spacing

I am wondering how to post-process the WRF output data into the format of example data (Example_input_OLR_model.nc)?

b) Are the thresholds in this case (parameters_features['threshold']=[250,225,200,175,150]) directly based on WRF's olr data (that is, Instantaneous Upwellng Longwave Flux at Top of Atmosphere)? Or do these thresholds refer to brightness temperature?

c) Is there currently an example on MCS tracking?

**Comment by ESSTOOL** [comment]
I tried reading the variable I want directly from the original WRF output file (using xarray), and then using xarry's to_iris(), but "dxy,dt=tobac.get_spacings(OLR)" still reports the same error. ..

**Comment by w-k-jones** [comment, marked as answer]
Hi, hope the following answers can provide some assistance:

a: There is a simple workaround to this, which is just to provide the values manually. dxy is the grid spacing in metres, and dt the time spacing in seconds. These values can either be passed directly to the detection and tracking functions, or provide the to "get spacings" function through the use of the "grid_spacing" and "time_spacing" keyword parameters (see here for more details: https://tobac.readthedocs.io/en/latest/tobac.html#tobac.utils.get_spacings)

For a more permanent solution, you could try using the WRFCube package to load your data, which was used to prepare the example data. It's available here: https://github.com/mheikenfeld/wrfcube

b: The threshold values apply to whatever variable is provided to the detection function (which is OLR in the case of the specific example). tobac is variable agnostic, so any data of your choice can be used to track clouds or other features. If you provided a brightness temperature field for feature detection a different choice of thresholds (e.g. [230, 220, 210, 200]) could be suitable

c: Not currently, but we are planning on adding more examples along with new features

**Comment by ESSTOOL** [comment]
Hi Jones,

Thanks for being so helpful!

Now I can use my own data for tracking of deep convection!

Also, I am very much looking forward to the launch of the MCS tracking example by tobac, because I think tobac is really useful!

Best,
Hongyong

**Comment by freemansw1** [comment]
@lk337 It sounds like this has been resolved; have you had success so far? If so, I'll close this issue.
